src/KeyManager.ts

```diff
@@ -1,13 +1,525 @@
-import kbpgp from 'kbpgp'
+import fs from 'fs-extra'
+import crypto from 'crypto'
+import chalk from 'chalk'
+import Path from 'path'
+import { promisify } from 'util'
+import { KeyPair } from './util'
 
-// we get back an object
-// not a series of modules I guess
-// otherwise the default is usually has to be import { default as ... } from ...
-// kbpgp.const.openpgp
-// so that's good
+// js imports
+const kbpgp = require('kbpgp')
+var F = kbpgp["const"].openpgp
+const zxcvbn = require('zxcvbn')
 
-class KeyManager {
+// TODO: remember that the symmetric keys will just be stored in vault directories.
+// I feel the symmetric key funciton here are not needed.
+// This should be dealing with asymm key management
+// Should the ciphering happen through cryptor?
+  // I think it should
 
+// Make a base class (or interaface) 'Key'. It has most of the methods you see here. The symetric and asymmetric keys
+// can both have their implementations of it. For example, they'd both have different generateKeyPair
+// Or maybe just return two different objects from within the class, one for symmetric and one for asymmetric
+
+// We want a consistent and uniform library. So even if we change crypto libraries for symm or asymm, not much
+// changes for compositional code. KeyManager, returns a 'Key' instance. There can be difference classes for both
+// symm and asymm. Cryptor can then take these Key classes, which expose the crytpo functions. So even if the
+// crypto library changes, Cryptor doesn't have to cahnge and neither does polykey.
+
+
+export default class KeyManager {
+  // TODO: wouldn't keymanager have many sym keys keys to look after?
+  private _keyPair: KeyPair = {private: '', public: '', passphrase: ''}
+  private _identity: Object | undefined = undefined
+  private _key!: Buffer
+  private _salt!: Buffer
+  private _passphrase!: string
+  private _storePath: string
+  private _fs: typeof fs
+  constructor(
```

Comment:
Can you write up a spec about what this class is supposed to do, and even a plantuml diagram that shows the interaction between these pieces.

For complex pieces of software like this, it's always better to start with a design spec in an issue or markdown file somewhere, so I can review according to the logic/spec, instead of just looking at syntax. It's like "reviewing in the small vs reviewing in the large".

Reply:
I have  put an outline of what the KeyManager class is supposed to do in the README